macros/src/main/scala/com/softwaremill/macwire/dependencyLookup/EligibleValuesFinder.scala

```diff
@@ -78,8 +78,12 @@ private[dependencyLookup] class EligibleValuesFinder[C <: blackbox.Context](val
           doFind(tail, values.put(scope, Ident(name), body))
 
         case ValDefOrEmptyDefDef(name, tpt, rhs, symbol) if name.toString != "<init>" =>
-          // rhs might be empty for local def
-          doFind(tail, values.put(scope, Ident(name), treeToCheck(tree, rhs)))
+          if(tpt.tpe == rhs.tpe) {
```

Comment:
Thanks for the PR! Maybe you could add a short comment here on why this check is needed (when the types are equal/distinct)?